### PR TITLE
feat(nuxt): Add Nuxt SDK docs

### PR DESCRIPTION
**Pull request by wobsoriano**
> [!IMPORTANT]
> 🔎 Previews:
> - https://clerk.com/docs/pr/1728/quickstarts/nuxt (Quickstart)
> - https://clerk.com/docs/pr/1728/quickstarts/overview#full-stack (Full stack entry)
> - https://clerk.com/docs/pr/1728/references/backend/overview
> - https://clerk.com/docs/pr/1728/references/nuxt/clerk-middleware (`clerkMiddleware` usage in Nuxt)
> - https://clerk.com/docs/pr/1728/references/nuxt/overview (Overview)
> - https://clerk.com/docs/pr/1728/references/nuxt/protecting-pages (Protecting pages)
> - https://clerk.com/docs/pr/1728/references/nuxt/read-session-data (Read session and user data)
> - https://github.com/clerk/clerk-nuxt-quickstart

#### Note:

I think it's best to review this PR after the Vue SDK PRs!

### Explanation:

- Alongside the new [Vue SDK](https://github.com/clerk/clerk-docs/pull/1709) we're launching, we're also launching an official Nuxt SDK. Nuxt is the fullstack framework for Vue.
- Nuxt SDK will be our fullstack SDK for Vue devs
- Here's the Nuxt SDK [PR](https://github.com/clerk/javascript/pull/4541), merged and `0.0.1` released

### This PR:

- Adds Nuxt SDK quickstart guide (quickstart repo [here](https://github.com/clerk/clerk-nuxt-quickstart), 1:1 with the guide)
- Ads SDK References and code examples

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

Resolves ECO-244

### CODE REVIEW

**Review by victoriaxyz — COMMENTED** [review]
Hi @wobsoriano, here's my [update](https://github.com/clerk/clerk-docs/pull/1728/commits/4fb15b5352bd4e36a6f226d8f66648fd8282e055)!

**docs/quickstarts/nuxt.mdx**
- update [quickstart](https://github.com/clerk/clerk-nuxt-quickstart/pull/1)
- fix `<TutorialHero/>` steps, links, headings
- rewrite and restructure sections

**docs/references/backend/overview.mdx**
- fix code file path and function name

**docs/references/nuxt/clerk-middleware.mdx**
- rewrite title, description, intro, sections
- rewrite and restructure to protecting routes via authentication and authorization status
- fix code logic, file paths, comments

**docs/references/nuxt/overview.mdx**
- rewrite title, description, intro, headings, sections

**docs/references/nuxt/protecting-pages.mdx**
- rewrite title, description, intro, headings, sections

**docs/references/nuxt/read-session-data.mdx**
- rewrite title, description, intro, headings, sections
- restructure sections
- fix code logic, file paths, comments

File: docs/references/nuxt/clerk-middleware.mdx [hunk 1]
```diff
@@ -64,53 +64,58 @@ export default clerkMiddleware((event) => {
...
+  const isAdminRoute = event.path.startsWith('/api/admin')
 
-  // Check if user is accessing sensitive customer data
-  if (isInvoicesRoute) {
-    // Check if user has the required permission
-    if (!canCreateInvoices) {
+  // Check if the user is trying to access a protected route
+  if (isAdminRoute) {
+    // Check if the user has the required admin role
+    if (orgRole !== 'org:admin') {
```
Comment (wobsoriano) [inline docs/references/nuxt/clerk-middleware.mdx:115]: I think we can also use  the `role` property here @alexisintech? Something lik,e `!has({ role: 'org:admin' })`

File: docs/references/nuxt/protect-pages.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+- [Use the `useAuth()` composable](#use-use-auth)
+- [Use `defineNuxtRouteMiddleware()`](#use-define-nuxt-route-middleware)
+
+> [!NOTE]
+> To learn how to protect API routes, see the [dedicated guide](/docs/references/nuxt/clerk-middleware#protect-api-routes).
+
+## Use `useAuth()`
+
+<Include src="_partials/nuxt/use-auth" />
```
Comment (wobsoriano) [inline docs/references/nuxt/protect-pages.mdx:16]: TIL 👀

File: docs/references/nuxt/clerk-middleware.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/references/nuxt/clerk-middleware.mdx:115]: yes, either orgRole or has({ role }). I can add an example

Comment (alexisintech) [reply docs/references/nuxt/clerk-middleware.mdx]: [c66fd2d](https://github.com/clerk/clerk-docs/pull/1728/commits/c66fd2d10ca2736b4c5b3c25eac93da8c4775c50)

**Pull request merged**